### ISSUE TITLE
CSS injected the speedy way sometimes is not split properly

### ISSUE THREAD

**Issue by piotrf** [styled-components/styled-components#1473]
## Version
styled-components: 3.1.4

## Reproduction

Here’s my gatsby sample live: https://priceless-minsky-70fda7.netlify.com
Here’s the the code for it: https://github.com/piotrf/gatsby-test

Also:

Here’s my live site using `"styled-components": "^3.0.2"`: https://piotrf.pl
And a dev build using `"styled-components": "^3.1.4"` https://develop.piotrf.pl

I made a little video comparing the two with the new version with FOUC on the right: https://s3-us-west-2.amazonaws.com/dropbox.piotrf.pl/20180131+styledcomponentsupdate.mov

## Steps to reproduce

Upgrading styled-components to 3.1.4 seemed to be enough for this to occur.

## Expected Behavior

No FOUC, content is being rendered smoothly.

## Actual Behavior

FOUC.

**Comment by probablyup** [comment]
It seems like an entire set of styles aren't being rendered during SSR... Could you try `styled-components@speedy`?

**Comment by probablyup** [comment]
Mostly curious to try a variant of 3.1.4 without the streaming changes to see if there might be a regression there.

**Comment by philpl** [comment]
@probablyup I think this might be because of how I implemented the rehydration? I have to replace the stylesheet before I add rules to it and it doesn't seem to be fast enough — just a hypothesis.

I could instead mount the new style tag after the old one and add all the rules, deleting the SSR'd one after.

**Comment by piotrf** [comment]
Trying `3.1.0-speedy6` now, will report back in a few.

**Comment by piotrf** [comment]
The issue persists with `speedy`.

**Comment by probablyup** [comment]
Ok that’s good to know, thank you.

**Comment by probablyup** [comment]
Ok, I think I know what's happening. @philpl it looks like the logic to split rules is a little broken.

Here's an example of a rule that s-c tried to inject:

```css
".fNLEFE a{font-size:18px;line-height:1.6em;}@media screen and (min-width:320px){.fNLEFE a{font-size:calc(18px + 6 * (100vw - 320px) / 1120);}}"
```

So I guess we aren't splitting on the braces correctly. Normal insertion would allow that, but insertRule won't.

**Comment by probablyup** [comment]
@piotrf I'm going to close the other issue and adjust this one to be representative of the current problem.

**Comment by mxstbr** [comment]
There is a full reproduction including a repo in #1454 (which I assume to be the same bug) in case somebody wants to tackle this!

**Comment by kitten** [comment]
cc @thysultan: the above comment looks like a bug in stylis-rule-plugin or stylis, I believe

**Comment by thysultan** [comment]
@kitten Can you open an issue with the repro on the stylis repo.

**Comment by kitten** [comment]
Will be fixed by upgrading stylis-rule-sheet in #1529
